app/apphandlers/gatewayAppHandler.go

```diff
@@ -129,11 +130,20 @@ func (a *gatewayApp) StartRudderCore(ctx context.Context, options *app.Options)
 		TransformerURL:           config.GetString("DEST_TRANSFORM_URL", "http://localhost:9090"),
 		FeaturesRetryMaxAttempts: 10,
 	})
+	drainConfigManager, err := drain_config.NewDrainConfigManager(config, a.log.Child("drain-config"))
+	if err != nil {
+		return fmt.Errorf("drain config manager setup: %v", err)
+	}
 	err = gw.Setup(
 		ctx,
 		config, logger.NewLogger().Child("gateway"), stats.Default,
 		a.app, backendconfig.DefaultBackendConfig, gatewayDB, errDB,
 		rateLimiter, a.versionHandler, rsourcesService, transformerFeaturesService, sourceHandle,
+		gateway.WithInternalHttpHandlers(
+			map[string]http.Handler{
+				"/drain-config": drainConfigManager.DrainConfigHttpHandler(),
```

Comment:
To cater for multinode setups and avoid having to perform adaptations in multinode-proxy, we may use the shared database for persisting such configurations, if available.